src/App.tsx

```diff
@@ -1,39 +1,52 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+
+type State = {
+  clockName: string;
+};
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
-
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: 'Clock-0',
+  };
+
+  nameChangeId = 0;
+
+  componentDidMount() {
+    this.nameChangeId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+    });
+  }
+
+  componentDidUpdate(
+    _prevProps: Readonly<State>,
+    prevState: Readonly<State>,
+  ): void {
```

Comment:
It's extra, bc TS already knows about React types
```suggestion
  componentDidUpdate(
    _prevProps: Readonly<State>,
    prevState: Readonly<State>,
  ) {
```